# Sample Proportions

Binomial distributions

$$\hat{p} = \frac{M}{N}$$

In fact, if your data $x_i$ is 1 for every "success" and 0 for every "failure", then we can say:

$$\hat{p} = \frac{1}{N} \sum_{i = 1}^{N} x_i = \bar{x}$$

That is, the sample proportion is the sample mean of the dataset.

# Confidence Interval for the Population Proportion

We have an interval with a lower and upper bound and we believe that the true population proportion is within this interval with some level of confidence. For a 95% confidence interval, we are "95% confident" the true proportion is in the interval (in the sense that such intervals contain the population proportion 95% of the time).

In a certain website, out of 1126 visitors on a given day, 310 clicked on an ad.

In [1]:
import statsmodels.api as sm

In [2]:
310 / 1126    # Sample proportion

0.2753108348134991

In [3]:
from statsmodels.stats.proportion import proportion_confint    # Function for computing confidence intervals
proportion_confint(count=310,    # Number of "successes"
                   nobs=1126,    # Number of trials
                   alpha=(1 - 0.95))    # Alpha, which is 1 minus the confidence level

(0.24922129423231776, 0.30140037539468045)

If we wanted a 99% confidence interval, we would have a wider interval, but more confidence that the true proportion lies in this interval.

In [4]:
proportion_confint(310, 1126, alpha=(1 - 0.99))

(0.24102336643386685, 0.30959830319313136)

## Lets Test the Proportion

We will do a **statistical test** to test the administrator's claim. We test the **null hypothesis**:
i
$$H_0: p = 0.3$$

(30% of the people visiting the website click the ad)

(where $p$ denotes the true proportion of visitors who click the ad on the site) against the **alternative hypothesis**:

$$H_A: p \neq 0.3$$

Compare against the p-value. If we see a $p$-value greater than 0.05, we will not reject the null hypothesis.

In [5]:
from statsmodels.stats.proportion import proportions_ztest    # Performs the test just described

res = proportions_ztest(count=310,
                        nobs=1126,
                        value=0.3,    # The hypothesized value of population proportion p
                        alternative='two-sided')

res    # A tuple; the first entry is the value of the test statistic, and the second is the p-value

(-1.8547614674673856, 0.06363029677684083)

Here, we got a test statistic of $z \approx -1.85$ and a $p$-value of $\approx 0.0636 > 0.05$. We conclude there is not enough statistical evidence to disagree with the website administrator.

## Now Common Proportions

Two different versions of an advertisement, Version A and Version B. Lets track how often Version A was clicked and how often Version B was clicked.

On a given day, 516 visitors saw Version A of the ad, and 510 saw Version B. Of those who saw Version A, 108 clicked the ad, while 144 clicked Version B when shown.

Which ad generates more clicks?

Here we test the following hypotheses:

$$H_0: p_A = p_B$$
$$H_A: p_A \neq p_B$$

perform `proportions_ztest()`

In [6]:
import numpy as np

In [7]:
np.array([108, 144])

array([108, 144])

In [8]:
proportions_ztest(count=np.array([108, 144]),
                  nobs=np.array([516, 510]),
                  alternative='two-sided')

(-2.7179204953199174, 0.0065693621488401655)

With a p-value of about 0.0066, which is small, we reject the null hypothesis; it appears that the two ads do not have the same proportion of clicks.

## $\chi^2$ or the Test for Goodness of Fit

The **$\chi^2$ test for goodness of fit** generalizes the test for a population proportion => checks whether a variable that could fall in some category has some distribution.

Suppose a website offers five colors of shoes: blue, black, brown, white, and red. We want to know whether each color is equally likely or not. That is, if $p_{\text{color}}$ is the proportion of shoe buyers who bought a particular color, we wish to test:

$$H_0: p_{\text{blue}} = p_{\text{black}} = p_{\text{brown}} = p_{\text{white}} = p_{\text{red}}$$
$$H_A: H_0 \text{ is false}$$

Suppose that out of 464 buyers of shoes, 98 bought blue shoes, 117 bought black shoes, 80 bought brown shoes, 73 bought white shoes, and 96 bought red shoes. If each shoe is equally likely to be bought, then $p_{\text{color}} = 0.2$ for every color. We would expect to see $0.2 \times 464 = 92.8$ pairs of each color sold if this were true.

In [9]:
from scipy.stats import chisquare

In [10]:
chisquare(f_obs=[98, 117, 80, 73, 96],    # Observed frequency for each color
          f_exp=[464 * .2, 464 * .2, 464 * .2, 464 * .2, 464 * .2])    # Expected frequency under the null hypothesis

Power_divergenceResult(statistic=12.70258620689655, pvalue=0.012824256359695625)

The p-value is approximately 0.0128. This is small, and suggests that the null hypothesis is false. It's likely that some shoe colors are more popular than others (black is a prime suspect).